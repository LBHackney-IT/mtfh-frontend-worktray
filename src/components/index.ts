export * from "./worktray-controls";
export * from "./worktray-filters";
export * from "./worktray-list";
export * from "./worktray-pagination";
export * from "./worktray-simple-list";
