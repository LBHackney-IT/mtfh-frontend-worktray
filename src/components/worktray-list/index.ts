export * from "./process-record";
export * from "./worktray-list";
export * from "./tenure-record";
