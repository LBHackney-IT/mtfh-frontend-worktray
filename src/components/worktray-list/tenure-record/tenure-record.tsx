import React from "react";

import { Process, RelatedEntity } from "@mtfh/common/lib/api/process/v1";
import { useTenure } from "@mtfh/common/lib/api/tenure/v1";
import { IProcess } from "@mtfh/processes";

import { ProcessRecord } from "../process-record";

interface TenureProcessRecordProps {
  process: Process;
  processConfig: IProcess;
  simple?: boolean;
}

export const TenureProcessRecord = (props: TenureProcessRecordProps): JSX.Element => {
  const { targetId, relatedEntities } = props.process;
  const relatedEntity = (relatedEntities as RelatedEntity[])?.[0];
  const { data: tenure } = useTenure(targetId);

  return (
    <ProcessRecord
      person={{
        id: relatedEntity?.id,
        fullName: relatedEntity?.description,
      }}
      property={{
        id: tenure?.tenuredAsset.id,
        address: tenure?.tenuredAsset.fullAddress,
      }}
      {...props}
    />
  );
};
