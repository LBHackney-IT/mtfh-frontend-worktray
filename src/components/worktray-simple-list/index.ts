export * from "./worktray-simple-list";
