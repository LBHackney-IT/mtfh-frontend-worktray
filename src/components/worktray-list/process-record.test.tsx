import React from "react";

import {
  getTenureV1,
  mockActiveTenureV1,
  mockPersonV1,
  mockProcessV1,
  render,
  server,
} from "@hackney/mtfh-test-utils";
import { screen } from "@testing-library/react";
import { subDays } from "date-fns";

import { Table } from "@mtfh/common/lib/components";
import { processes } from "@mtfh/processes";

import { TenureProcessRecord } from "./tenure-record";

const { states } = processes.soletojoint;
const mockProcess = {
  ...mockProcessV1,
  relatedEntities: [
    {
      id: mockPersonV1.id,
      description: `${mockPersonV1.firstName} ${mockPersonV1.surname}`,
      targetType: "person",
      subType: "",
    },
  ],
};

describe("process-record-component", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders TenureProcessRecord correctly", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    const { container } = render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "BreachChecksPassed",
              createdAt: new Date().toISOString(),
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(
      screen.findByText(mockActiveTenureV1.tenuredAsset.fullAddress),
    ).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.breachChecksPassed.status);
    expect(status.className).toContain("grey");
    expect(container).toMatchSnapshot();
  });

  test("it renders TenureProcessRecord correctly, 1 day left", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "BreachChecksPassed",
              createdAt: subDays(new Date(), 9).toISOString(),
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(
      screen.findByText(mockActiveTenureV1.tenuredAsset.fullAddress),
    ).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.breachChecksPassed.status);
    expect(status.className).toContain("amber");
  });

  test("it renders TenureProcessRecord correctly, 5 days left", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "BreachChecksPassed",
              createdAt: subDays(new Date(), 5).toISOString(),
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(
      screen.findByText(mockActiveTenureV1.tenuredAsset.fullAddress),
    ).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.breachChecksPassed.status);
    expect(status.className).toContain("aqua");
  });

  test("it renders TenureProcessRecord correctly, overdue", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "BreachChecksPassed",
              createdAt: subDays(new Date(), 12).toISOString(),
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(
      screen.findByText(mockActiveTenureV1.tenuredAsset.fullAddress),
    ).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.breachChecksPassed.status);
    expect(status.className).toContain("red");
  });

  test("it renders TenureProcessRecord correctly, state=ProcessClosed", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "ProcessClosed",
              createdAt: "2022-01-01T00:00:00Z",
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(screen.findByText("-")).resolves.toBeInTheDocument();
    await expect(screen.findByText("Closed")).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.processClosed.status);
    expect(status.className).toContain("orange");
  });

  test("it renders TenureProcessRecord correctly, state=ProcessCancelled", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "ProcessCancelled",
              createdAt: "2022-01-01T00:00:00Z",
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(screen.findByText("-")).resolves.toBeInTheDocument();
    await expect(screen.findByText("Cancelled")).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.processCancelled.status);
    expect(status.className).toContain("orange");
  });

  test("it renders TenureProcessRecord correctly, state=TenureUpdated", async () => {
    server.use(getTenureV1(mockActiveTenureV1, 200));
    render(
      <Table>
        <TenureProcessRecord
          process={{
            ...mockProcess,
            currentState: {
              ...mockProcess.currentState,
              state: "TenureUpdated",
              createdAt: "2022-01-01T00:00:00Z",
            },
          }}
          processConfig={processes.soletojoint}
        />
      </Table>,
    );
    await expect(screen.findByText("-")).resolves.toBeInTheDocument();
    await expect(screen.findByText("Completed")).resolves.toBeInTheDocument();
    const status = await screen.findByText(states.tenureUpdated.status);
    expect(status.className).toContain("green");
  });
});
