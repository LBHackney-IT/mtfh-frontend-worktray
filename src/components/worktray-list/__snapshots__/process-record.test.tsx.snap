// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process-record-component it renders PersonProcessRecord correctly 1`] = `
<div>
  <table
    class="govuk-table lbh-table"
  >
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row govuk-table__row process-record"
      >
        <td
          class="govuk-table__cell process-record__item"
        >
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
            href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
            target="_blank"
          >
            Haylee Macejkovic
          </a>
        </td>
        <td
          class="govuk-table__cell process-record__item --processName"
        >
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
            href="/processes/ab/6fbe024f-2316-4265-a6e8-d65a837e308a"
            target="_blank"
          >
            Change of Name
          </a>
        </td>
        <td
          class="govuk-table__cell process-record__item"
        >
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
            href="/"
            target="_blank"
          >
            CP*
          </a>
        </td>
        <td
          class="govuk-table__cell process-record__item"
        >
          <p
            class="lbh-body-m title"
          >
            Initiated
          </p>
          <p
            class="lbh-body-m createdAt"
          >
            20/08/2022
          </p>
        </td>
        <td
          class="govuk-table__cell process-record__item"
        >
          <p
            class="lbh-body-m"
          >
            -
          </p>
        </td>
        <td
          class="govuk-table__cell process-record__item --status"
        >
          <span
            class="govuk-tag lbh-tag box aqua"
          >
            Awaiting supporting documents through DES
          </span>
          <p
            class="lbh-body-s"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`process-record-component it renders TenureProcessRecord correctly 1`] = `
<div>
  <table
    class="govuk-table lbh-table"
  >
    <tbody
      class="govuk-table__body"
    >
      <tr
        class="govuk-table__row govuk-table__row process-record"
      >
        <td
          class="govuk-table__cell process-record__item"
        >
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
            href="/person/6fbe024f-2316-4265-a6e8-d65a837e308a"
            target="_blank"
          >
            Haylee Macejkovic
          </a>
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state --address"
            href="/property/f69ae994-4948-481f-a2d8-d52cfcb68321"
            target="_blank"
          >
            72251 Champlin Trafficway, 99647
          </a>
        </td>
        <td
          class="govuk-table__cell process-record__item --processName"
        >
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
            href="/processes/ab/6fbe024f-2316-4265-a6e8-d65a837e308a"
            target="_blank"
          >
            Sole to Joint
          </a>
        </td>
        <td
          class="govuk-table__cell process-record__item"
        >
          <a
            class="govuk-link lbh-link lbh-link--no-visited-state lbh-heading-h4"
            href="/"
            target="_blank"
          >
            CP*
          </a>
        </td>
        <td
          class="govuk-table__cell process-record__item"
        >
          <p
            class="lbh-body-m title"
          >
            Initiated
          </p>
          <p
            class="lbh-body-m createdAt"
          >
            20/08/2022
          </p>
        </td>
        <td
          class="govuk-table__cell process-record__item"
        >
          <p
            class="lbh-body-m"
          >
            10 days left
          </p>
        </td>
        <td
          class="govuk-table__cell process-record__item --status"
        >
          <span
            class="govuk-tag lbh-tag box grey"
          >
            Request supporting documents
          </span>
          <p
            class="lbh-body-s"
          >
            10 days left
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
